src/wp-includes/class-wp-metadata-lazyloader.php

```diff
@@ -187,6 +191,10 @@ public function lazyload_meta_callback( $check, $object_id, $meta_key, $single,
 
 		$object_ids = array_keys( $this->pending_objects[ $meta_type ] );
 		if ( $object_id && ! in_array( $object_id, $object_ids, true ) ) {
+			// Only load in the post meta value is in the queue.
+			if ( 'post' === $meta_type ) {
+				return $check;
+			}
```

Comment:
@spacedmonkey Re https://github.com/WordPress/wordpress-develop/pull/4216#discussion_r1966428938, I think the code should simply be this to see if a database query is required. If not, it bypasses processing the queue, if so it adds the object ID and continues lazy loading.

```suggestion
			if ( false !== wp_cache_get( $object_id, $meta_type . '_meta' ) ) {
				// Object already in cache, no database query will be made.
				return $check;
			}
```

Reply:
@peterwilsoncc This happens in here 
https://github.com/WordPress/wordpress-develop/blob/bcb59bfb14cbfa79335fc9e411aae56b060da85d/src/wp-includes/meta.php#L1168-L1189

So if it first checks if the id is in cache, then if not, then it will do a database query. 

Add this check, just repeats to the logic.

Reply:
It's not quite the same.

`get_metadata( type, **cached** object ID ...)` will process the queue, even though a database query is not required. This is because ` "get_{$meta_type}_metadata"` fires prior to checking the objects cache.

As you've been trying to avoid unnecessary database queries while getting post meta data, this acheives that by only processing the queue if a database query will be made.